Indicators/ChandeKrollStop.cs

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data.Market;
+using System;
+using System.Linq;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// This indicator computes the short stop and lower stop values of the Chande Kroll Stop Indicator.
+    /// It is used to determine the optimal placement of a stop-loss order.
+    /// </summary>
+    public class ChandeKrollStop : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        private readonly AverageTrueRange _atr;
+        private readonly decimal _atrMult;
+        private readonly RollingWindow<IBaseDataBar> _inputValues;
+        private readonly RollingWindow<decimal> _high_stop_list;
+        private readonly RollingWindow<decimal> _low_stop_list;
+
+        /// <summary>
+        /// Gets the short stop of ChandeKrollStop.
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> ShortStop { get; }
+
+        /// <summary>
+        /// Gets the long stop of ChandeKrollStop.
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> LongStop { get; }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples > 0;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChandeKrollStop"/> class.
+        /// </summary>
+        /// <param name="atrPeriod">The period over which to compute the average true range.</param>
+        /// <param name="atrMult">The ATR multiplier to be used to compute stops distance.</param>
+        /// <param name="period">The period over which to compute the max of high stop and min of low stop.</param>
+        public ChandeKrollStop(int atrPeriod, decimal atrMult, int period)
+            : this($"CKS({atrPeriod},{atrMult},{period})", atrPeriod, atrMult, period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChandeKrollStop"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="atrPeriod">The period over which to compute the average true range.</param>
+        /// <param name="atrMult">The ATR multiplier to be used to compute stops distance.</param>
+        /// <param name="period">The period over which to compute the max of high stop and min of low stop.</param>
+        public ChandeKrollStop(string name, int atrPeriod, decimal atrMult, int period)
+            : base(name)
+        {
+            WarmUpPeriod = 1;
```

Comment:
> WarmUpPeriod = 1;

Does not seem right? Strange that tests didn't catch this, there might be some other bug there

Reply:
You are right this shouldn't be 1. With `ChandeKrollStop(5, 2.0m, 3);` we are only having ouputs after 7 incoming data but if I put